src/app/admin/results/result-groups-raiting/result-groups-raiting.component.ts

```diff
@@ -0,0 +1,126 @@
+import { Component, OnInit, Inject } from '@angular/core';
+import { ResultsService } from '../results.service';
+import { MatDialogRef, MAT_DIALOG_DATA} from '@angular/material';
+import { Results, Student, Group } from 'src/app/shared/entity.interface';
+import { ApiService } from 'src/app/shared/services/api.service';
+
+@Component({
+  selector: 'app-result-groups-raiting',
+  templateUrl: './result-groups-raiting.component.html',
+  styleUrls: ['./result-groups-raiting.component.scss']
+})
+export class ResultGroupsRaitingComponent implements OnInit {
+
+  public groupNames = [];
+  public score = [];
+  public chardReady = false;
+  public isLoading = true;
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public data: any,
+    public dialogRef: MatDialogRef<ResultGroupsRaitingComponent>,
+    private resultsService: ResultsService,
+    private apiService: ApiService,
+  ) { }
+
+  public barChartLabels: string[];
+  public barChartType  = 'bar';
+  public barChartLegend = true;
+  public barChartData;
+  public barChartOptions = {
+    scaleShowVerticalLines: true,
+    responsive: true,
+    scales: {
+      yAxes: [{
+        gridLines: {
+          zeroLineColor: 'black',
+          zeroLineWidth: 1
+        },
+        ticks: {
+          min: 0,
+          max: 100
+        },
+        scaleLabel: {
+          display: true,
+          labelString: 'Успішність студента,(%)'
+        }
+      }],
+      xAxes: [{
+        scaleLabel: {
+          display: true,
+          labelString: `Ім'я студента`
+        }
+      }]
+    }
+  };
+
+  ngOnInit() {
+    this.getDataForChart();
+  }
+
+  private getDataForChart() {
+    this.getScoreForChart();
+    this.getGroupNamesForChart();
+  }
+
+  private async getTestResultsList() {
+    const promise: any = this.resultsService.getRecordsByTestGroupDate(this.data.testID).toPromise();
+    return Promise.resolve(promise);
+  }
+
+  private async getStudentsList() {
+    const testResultsList = [...new Set(await this.getTestResultsList())];
```

Comment:
What is the reason of using Set?

Reply:
In order to leave just unique items in  array